stride/physics/iso_elastic/devito.py

```diff
@@ -0,0 +1,398 @@
+
+    # forward
+
+    async def before_forward(self, wavelets, vp, vs, rho, **kwargs):
+        """
+        Prepare the problem type to run the state or forward problem.
+
+        Parameters
+        ----------
+        wavelets : Traces
+            Source wavelets.
+        vp : ScalarField
+            Compressional (acoustic) speed of sound of the medium, in [m/s].
+        vs : ScalarField
+            Transverse (shear) speed of sound of the medium, in [m/s].
+        rho : ScalarField
+            Density of the medium in [kg/m^3].
+        problem : Problem
+            Sub-problem being solved by the PDE.
+        save_wavefield : bool or int, optional
+            Whether or not to solve the forward wavefield, defaults to True when
+            a gradient is expected, and to False otherwise. An integer number N can
+            also be provided, in which case the last N timesteps of the wavefield
+            will be saved.
+        boundary_type : str, optional
+            Type of boundary for the wave equation (``sponge_boundary_2`` or
+            ``complex_frequency_shift_PML_2``), defaults to ``sponge_boundary_2``.
+        interpolation_type : str, optional
+            Type of source/receiver interpolation (``linear`` or ``hicks``), defaults
+            to ``linear``.
+        attenuation_power : int, optional
+            Power of the attenuation law if attenuation is given (``0`` or ``2``),
+            defaults to ``0``.
+        drp : bool, optional
+            Whether or not to use dispersion-relation preserving coefficients (only
+            available in some versions of Stride). Defaults to False.
+        kernel : str, optional
+            Type of time kernel to use (``OT2`` for 2nd order in time or ``OT4`` for 4th
+            order in time). If not given, it is automatically decided given the time spacing.
+        undersampling : int, optional
+            Amount of undersampling in time when saving the forward wavefield. If not given,
+            it is calculated given the bandwidth.
+
+
+        Returns
+        -------
+
+        """
+        problem = kwargs.get('problem')
+        shot = problem.shot
+
+        num_sources = shot.num_sources
+        num_receivers = shot.num_receivers
+
+        # If there's no previous operator, generate one
+        if self.state_operator.devito_operator is None:
+
+            # Define variables
+            src = self.dev_grid.sparse_time_function('src', num=num_sources,
+                                                     coordinates=shot.source_coordinates,
+                                                     interpolation_type=self.interpolation_type)
+            rec_tau = self.dev_grid.sparse_time_function('rec_tau', num=num_receivers,
+                                                     coordinates=shot.receiver_coordinates,
+                                                     interpolation_type=self.interpolation_type)
+
+            s = self.time.step
```

Comment:
use "step" instead of "s", it is more descriptive

Reply:
Correction made